sound/hda/ext/hdac_ext_controller.c

```diff
@@ -225,9 +225,7 @@ int snd_hdac_ext_bus_link_power_up_all(struct hdac_bus *bus)
 	int ret;
 
 	list_for_each_entry(hlink, &bus->hlink_list, list) {
-		snd_hdac_updatel(hlink->ml_addr, AZX_REG_ML_LCTL,
-				 AZX_ML_LCTL_SPA, AZX_ML_LCTL_SPA);
-		ret = check_hdac_link_power_active(hlink, true);
+		ret = snd_hdac_ext_bus_link_power_up(hlink);
```

Comment:
Optionally you can use `snd_hdac_ext_bus_link_power_up_all()`

and

Reply:
sorry, not getting the feedback @ujfalusi . This is the snd_hdac_ext_bus_link_power_up_all() routine, I can't call it recursively?

Reply:
Aha, oh, yeah, I should have looked up in the diff, sorry!